sys/posix/sockets/posix_sockets.c

```diff
@@ -465,8 +464,9 @@ int connect(int socket, const struct sockaddr *address, socklen_t address_len)
     switch (s->type) {
 #ifdef MODULE_CONN_TCP
         case SOCK_STREAM:
-            /* XXX sin_port is in network byteorder */
-            if ((res = conn_tcp_connect(&s->conn.tcp, addr, addr_len, port)) < 0) {
+            if ((res = conn_tcp_connect(&s->conn.tcp, addr, addr_len,
+                                        byteorder_ntohs(port))) < 0) {
```

Comment:
Ah, I see, in POSIX sockaddrs they're network byte order, but RIOT's interface is from this century, so we have to convert.